### PR TITLE
libSyntax: specialize class declaration syntax node.

### PR DESCRIPTION
**Pull request by ftchirou**
This PR specializes class declaration syntax nodes.

Resolves [SR-6571](https://bugs.swift.org/browse/SR-6571).

### CODE REVIEW

File: test/Syntax/Inputs/serialize_class_decl.json
Comment (harlanhaskins) [inline test/Syntax/Inputs/serialize_class_decl.json]: I actually don’t think we need this test. I’d like to remove these JSON tests because they create a lot of churn in the test suite — I’m gonna replace the existing ones with FileCheck based tests that test more structural properties of the JSON.

**Review by harlanhaskins — CHANGES_REQUESTED** [review]
Patch LGTM modulo the test. Would be nice to add a case to the existing test that includes attributes and inheritance clauses.

**Review by harlanhaskins — APPROVED** [review]
This looks great! Thanks for tackling this!

**Comment by harlanhaskins** [comment]
@swift-ci please test

File: test/Syntax/round_trip_parse_gen.swift
Comment (nkcsgexi) [inline test/Syntax/round_trip_parse_gen.swift]: could we also add a test using decl attributes? like:
```
@available(*, unavailable)
private class C {}
```

**Review by nkcsgexi — COMMENTED** [review]
Looking good! 👍  mind adding one more test?

File: test/Syntax/round_trip_parse_gen.swift
Comment (ftchirou) [reply test/Syntax/round_trip_parse_gen.swift]: Yup. Done.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@ftchirou Some other comments:
- can we update lib/Syntax/Status.md indicating ClassDecl is done? 
- can we squash these commits before merging?

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by nkcsgexi — APPROVED** [review]

**Comment by nkcsgexi** [comment]
🚢

**Pull request merged**
